### PR TITLE
fix installation of Bioconductor 3.11 bundle on aarch64

### LINKED ISSUE #11393: R-bundle-Bioconductor-3.11-foss-2020a-R-4.0.0.eb failures on aarch64.
There are multiple packages in the Bioconductor suite that fail on aarch64. This is now a complete list. 

Extensions:
- [x] dada2: https://github.com/benjjneb/dada2/pull/1147 (that fix requires fixing, I'll try to fix that...)
- [x] LEA: (depends on dada2, does not support anything but `__i386__` and `__x86_64__` due to https://github.com/bcm-uga/LEA/blob/master/src/bituint/bituint.h)
- [x] mzR: (uses a partial boost done at [home] (https://github.com/sneumann/mzR/blob/beb109476546d58a903eacd3d263e07f94d35a58/src/README.updateboost)). Fixed with using external boost via patch by @zao.
- [x] MSnbase (depends on mzR)
- [x] xcms (depends on mzR)
- [x] CAMERA (depends on xcms)
- [x] flowClust (Uses a local old version of GSL: https://github.com/RGLab/flowClust/issues/5, fails with `undefined symbol: gsl_min_fminimizer_brent`) - Fixed with external GSL: https://github.com/easybuilders/easybuild-easyconfigs/pull/11417/commits/4338a18f90a55c2f634950606cff504bd6f6cdae
- [x] openCyto (depends on flowClust)
- [x] CytoML (depends on openCyto)
- [x] FlowSOM (depends on CytoML)

Other:

- [x] Fix building with optarch=GENERIC (https://github.com/easybuilders/easybuild-easyblocks/issues/2195)

### PR DESCRIPTION
**Pull request by terjekv**
(created using `eb --new-pr`)

Eventually fixes #11393

### CODE REVIEW

**Comment by terjekv** [comment]
Test report by @terjekv
**FAILED**
Build succeeded for 6 out of 7 (1 easyconfigs in this PR)
ninhursaga.uio.no - Linux RHEL 8.2, x86_64, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz (cascadelake), Python 3.6.8
See https://gist.github.com/cd491e68799cbbb3cc2a97c377c8431d for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by terjekv** [comment]
Test report by @terjekv
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
ninhursaga.uio.no - Linux RHEL 8.2, x86_64, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz (cascadelake), Python 3.6.8
See https://gist.github.com/3b8c20372e1a17fde286f81e2645f7da for a full test report.

**Comment by terjekv** [comment]
Test report by @terjekv
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
ninhursaga.uio.no - Linux RHEL 8.2, x86_64, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz (cascadelake), Python 3.6.8
See https://gist.github.com/d78180e3380421bc2fad685204288bb7 for a full test report.

**Comment by terjekv** [comment]
Test report by @terjekv
**FAILED**
Build succeeded for 2 out of 3 (1 easyconfigs in this PR)
arm1 - Linux ubuntu 18.04, AArch64, UNKNOWN (aarch64), Python 3.6.9
See https://gist.github.com/9f46ac2a6e606970807ee4b5fbc8e8bb for a full test report.

**Comment by terjekv** [comment]
Test report by @terjekv
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
ninhursaga.uio.no - Linux RHEL 8.2, x86_64, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz (cascadelake), Python 3.6.8
See https://gist.github.com/478a3bfe3f6e1f02c97350f19c3e41b9 for a full test report.

**Comment by terjekv** [comment]
Test report by @terjekv
**SUCCESS**
Build succeeded for 0 out of 0 (1 easyconfigs in this PR)
ninhursaga.uio.no - Linux RHEL 8.2, x86_64, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz (cascadelake), Python 3.6.8
See https://gist.github.com/3a35d60ac69fa8b76d66c9b0fcef48c7 for a full test report.

**Comment by terjekv** [comment]
Test report by @terjekv
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
ninhursaga.uio.no - Linux RHEL 8.2, x86_64, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz (cascadelake), Python 3.6.8
See https://gist.github.com/5c9abc6009600e556df09bb6f0d8d779 for a full test report.

**Comment by terjekv** [comment]
Test report by @terjekv
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
arm1 - Linux ubuntu 18.04, AArch64, UNKNOWN (aarch64), Python 3.6.9
See https://gist.github.com/d1ad930dcce6014d01990ff2069b0e17 for a full test report.

**Comment by terjekv** [comment]
Test report by @terjekv
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
arm1 - Linux ubuntu 18.04, AArch64, UNKNOWN (aarch64), Python 3.6.9
See https://gist.github.com/3fab57e774c4142bd2e89eff77c30b22 for a full test report.

**Comment by terjekv** [comment]
Test report by @terjekv
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
ninhursaga.uio.no - Linux RHEL 8.2, x86_64, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz (cascadelake), Python 3.6.8
See https://gist.github.com/cb778007bb67dea0ba613138d2abb517 for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 3 out of 3 (1 easyconfigs in this PR)
bear-pg0305u03a.bear.cluster - Linux RHEL 7.6, POWER, 8335-GTX (power9le), Python 3.6.8
See https://gist.github.com/c0a1a6f49a9c06542d73ad5e5aed3a84 for a full test report.

**Comment by terjekv** [comment]
Both of the aarch64 units I have access to are currently down. I'll try to have someone look at it tomorrow, but last I saw the testing had long since past LEA and other troublemakers:

```
[...]
== installing extension IlluminaHumanMethylation450kmanifest 0.4.0 (242/262)...
== installing extension IlluminaHumanMethylationEPICmanifest 0.3.0 (243/262)...
== installing extension IlluminaHumanMethylation450kanno.ilmn12.hg19 0.6.0 (244/262)...
client_loop: send disconnect: Broken pipe
```

**Comment by terjekv** [comment]
No longer a WIP. This looks good. Testing on arm again today.

**Comment by boegel** [comment]
@terjekv Is a test report coming up?

**Comment by terjekv** [comment]
Yeah, I thought I started it last night. Sorry. Let me kick one off.

**Comment by terjekv** [comment]
Test report by @terjekv
**FAILED**
Build succeeded for 77 out of 97 (1 easyconfigs in this PR)
arm2 - Linux ubuntu 18.04, AArch64, UNKNOWN, Python 3.6.9
See https://gist.github.com/7b023efdccc8dfdadc95ee6b07af6420 for a full test report.

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
easybuild.novalocal - Linux centos linux 7.8.2003, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 2.7.5
See https://gist.github.com/e3bdb1e4dd751df521669055460eb686 for a full test report.

**Comment by terjekv** [comment]
> Test report by @verdurin
> **FAILED**
> Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
> easybuild.novalocal - Linux centos linux 7.8.2003, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 2.7.5
> See https://gist.github.com/e3bdb1e4dd751df521669055460eb686 for a full test report.

@verdurin This has to be built against latest Develop. It's the same issue I ran into, as underlying EasyConfigs have changed.

**Comment by terjekv** [comment]
Test report by @terjekv
**FAILED**
Build succeeded for 0 out of 20 (1 easyconfigs in this PR)
arm2 - Linux ubuntu 18.04, AArch64, UNKNOWN, Python 3.6.9
See https://gist.github.com/968cdc2438cc87d34d5eac22d1867b17 for a full test report.

**Comment by terjekv** [comment]
> Test report by @terjekv
> **FAILED**
> Build succeeded for 0 out of 20 (1 easyconfigs in this PR)
> arm2 - Linux ubuntu 18.04, AArch64, UNKNOWN, Python 3.6.9
> See https://gist.github.com/968cdc2438cc87d34d5eac22d1867b17 for a full test report.

Sigh, `unzip` is a system dependency for Python:

```
== FAILED: Installation ended unsuccessfully (build directory: /home/terjekv/.local/easybuild/build/Python/3.8.2/GCCcore-9.3.0): build failed (first 300 chars): cmd "unzip -qq /h
ome/terjekv/.local/easybuild/sources/p/Python/extensions/setuptools-45.2.0.zip" exited with exit code 127 and output:
/bin/bash: unzip: command not found
```

**Comment by terjekv** [comment]
Test report by @terjekv
**FAILED**
Build succeeded for 17 out of 20 (1 easyconfigs in this PR)
arm2 - Linux ubuntu 18.04, AArch64, UNKNOWN, Python 3.6.9
See https://gist.github.com/a2b1e3300c90884886f8e4064e4110a2 for a full test report.

**Comment by terjekv** [comment]
Test report by @terjekv
**SUCCESS**
Build succeeded for 3 out of 3 (1 easyconfigs in this PR)
arm2 - Linux ubuntu 18.04, AArch64, UNKNOWN, Python 3.6.9
See https://gist.github.com/b80ca5c715108accc49be382f3062b4e for a full test report.

**Comment by boegel** [comment]
> > Test report by @terjekv
> > **FAILED**
> > Build succeeded for 0 out of 20 (1 easyconfigs in this PR)
> > arm2 - Linux ubuntu 18.04, AArch64, UNKNOWN, Python 3.6.9
> > See https://gist.github.com/968cdc2438cc87d34d5eac22d1867b17 for a full test report.
> 
> Sigh, `unzip` is a system dependency for Python:
> 
> ```
> == FAILED: Installation ended unsuccessfully (build directory: /home/terjekv/.local/easybuild/build/Python/3.8.2/GCCcore-9.3.0): build failed (first 300 chars): cmd "unzip -qq /h
> ome/terjekv/.local/easybuild/sources/p/Python/extensions/setuptools-45.2.0.zip" exited with exit code 127 and output:
> /bin/bash: unzip: command not found
> ```

We have easyconfigs for `UnZip`, so we should add it as a build dep for `Python`, see #10477.

**Comment by terjekv** [comment]
Test report by @terjekv
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
arm2 - Linux ubuntu 18.04, AArch64, UNKNOWN, Python 3.6.9
See https://gist.github.com/6919d7d54c31e11e0a10c26b641b30c3 for a full test report.

File: easybuild/easyconfigs/r/R-bundle-Bioconductor/mzR-2.22.0_use-external-boost.patch [hunk 1]
```diff
@@ -0,0 +1,129 @@
+diff -ru mzR.orig/src/Makevars mzR/src/Makevars
```
Comment (boegel) [inline easybuild/easyconfigs/r/R-bundle-Bioconductor/mzR-2.22.0_use-external-boost.patch:1]: @terjekv missing patch description + author or upstream reference

File: easybuild/easyconfigs/r/R-bundle-Bioconductor/LEA-3.0.0_support_aarch64_and_ppc64le.patch [hunk 2]
```diff
@@ -0,0 +1,60 @@
+diff --git a/src/bituint/bituint.h b/src/bituint/bituint.h
```
Comment (boegel) [inline easybuild/easyconfigs/r/R-bundle-Bioconductor/LEA-3.0.0_support_aarch64_and_ppc64le.patch:1]: @terjekv missing patch description + author or upstream reference

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/r/R-bundle-Bioconductor/LEA-3.0.0_support_aarch64_and_ppc64le.patch (hunk 2 above)
Comment (terjekv) [reply easybuild/easyconfigs/r/R-bundle-Bioconductor/LEA-3.0.0_support_aarch64_and_ppc64le.patch:7]: https://github.com/easybuilders/easybuild-easyconfigs/pull/11417/commits/23c4a7616e4b37d7c6e4e91d6f52fae06c743cb9

File: easybuild/easyconfigs/r/R-bundle-Bioconductor/mzR-2.22.0_use-external-boost.patch (hunk 1 above)
Comment (terjekv) [reply easybuild/easyconfigs/r/R-bundle-Bioconductor/mzR-2.22.0_use-external-boost.patch:5]: https://github.com/easybuilders/easybuild-easyconfigs/pull/11417/commits/23c4a7616e4b37d7c6e4e91d6f52fae06c743cb9

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3407.kirlia.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz (cascadelake), Python 2.7.5
See https://gist.github.com/2452f534a8feed0ce045ab283aec22c6 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @terjekv!

**Pull request merged**